static/js/containers/DashboardPage.js

```diff
@@ -325,24 +343,29 @@ class DashboardPage extends React.Component {
       errorMessage = <ErrorMessage errorInfo={{user_message: "No program enrollment is available."}} />;
     } else {
       let financialAidCard;
-      if (program.financial_aid_availability) {
-        financialAidCard = <FinancialAidCard
-          program={program}
-          coursePrice={coursePrice}
-          openFinancialAidCalculator={this.openFinancialAidCalculator}
-          documents={documents}
-          setDocumentSentDate={this.setDocumentSentDate}
-          skipFinancialAid={this.skipFinancialAid}
-          updateDocumentSentDate={this.updateDocumentSentDate}
-          setConfirmSkipDialogVisibility={this.setConfirmSkipDialogVisibility}
-          setDocsInstructionsVisibility={this.setDocsInstructionsVisibility}
-          ui={ui}
-          financialAid={financialAid}
-        />;
-      }
 
       const calculatedPrices = calculatePrices(dashboard.programs, prices.coursePrices, coupons.coupons);
 
+      if ( program.financial_aid_availability ) {
+        if ( calculatedPrices.get(program.id) === 0 ) {
```

Comment:
hm, I think the latter might be a better solution - if someone has a 100% for each course we wouldn't want to show the financial aid card, right?